File: sort/selection.py
"""###  DSA ###"""
def selection_sort(arr:list):
    """
    Selection sort alogorithm implrmrntaion in python.
    :params arr:List of Integers
    :return List of Integers
    """
    length=len(arr)
    for i in range(length):
        min_index=i
        for j in range(i+1,length):
            if arr[j]< arr[min_index]:
                min_index=j
        arr[i],arr[min_index]=arr[min_index],arr[i]
    return arr

 #example
array=[12,32,1,45,321,4]
print(selection_sort(array))

File: sort/selection.cpp
#include <iostream>
#include <vector>

using namespace std;

void selection_sort(vector<int>&arr){
    for (int i =0;i<arr.size();i++){
        int min_index= i;
        for (int j=i+1;j<arr.size();j++){
            if (arr[j]< arr[min_index]){
               min_index=j;
            }
        }
        swap(arr[i],arr[min_index]);
    }
}
int main(){

    vector<int> arr = {5,3,8,1,23,43,221};
    selection_sort(arr);
    for (int i =0; i<arr.size(); i++){
        cout << arr[i] << " ";
    }

    cout << endl;

    return 0;
}